assets/dev/js/frontend/handlers/container/grid-container.js

```diff
@@ -273,4 +277,56 @@ export default class GridContainer extends elementorModules.frontend.handlers.Ba
 
 		return ! numberPattern.test( gridRows?.size );
 	}
+
+	shouldRemoveEmptyView() {
+		const childrenLength = this.elements.outlineParentContainer.querySelectorAll( ':scope > .elementor-element' ).length;
+
+		if ( 0 === childrenLength ) {
+			return false;
+		}
+
+		const maxElements = this.getMaxElementsNumber();
+
+		return maxElements <= childrenLength && this.isFullFilled( childrenLength );
+	}
+
+	getMaxElementsNumber() {
```

Comment:
I would move this function to the end as I am using it later on as well. 
Just for the file order...